cips/cip-x.md

```diff
@@ -0,0 +1,77 @@
+| cip | XX (assigned by Editors) |
+| - | - |
+| title | Header Pruning for Light Nodes |
+| description | Mechanism to retain a fixed range of headers instead of whole history |
+| author | Hlib Kanunnikov ([@Wondertan](https://github.com/Wondertan)) |
+| discussions-to | <https://forum.celestia.org/t/cip-header-pruning-for-lns/1958> |
+| status | Draft |
+| type | Standards Track |
+| category | Data Availability |
+| created | 2025-03-21 |
+
+## Abstract
+
+Currently, every data availability (DA) node type synchronizes all historical headers starting from genesis (or other statically configured
+historical header) until the subjectively initialized head of the chain. We change that by adding a way to sync a
+constant size range of headers instead of the whole history.
+
+## Motivation
+
+Light nodes (LNs) store the whole history of ExtendedHeaders in their entirety, including ValidatorSet, Commit and data availability header (DAH). The motivation
+is to decrease LN storage requirements by retaining only a configurable window of recent headers.
+
+## Specification
+
+In order to reference the last stored header, we introduce a notion of **Tail** header of the chain, complementing
+**Head** header. Tail header height is estimated with [HeaderPruningWindow](#parameters), block time and Head height.
+
+- On the first start of a node, the Tail is retrieved from trusted peers alongside with the Head, performing subjective
+initialization.
+  - Once retrieved, Tail becomes the point of trust for the initializing node where everything beforehand assumed valid
+ (unless syncing from genesis)
+  - Head, on the other hand, is not trusted and becomes a synchronization target for the node.
```

Comment:
if Head was retrieved in the exact same manner as Tail as part of subjective init, and Tail is trusted, what's the justification for not trusting Head? I would have expected the latest height that is subjectively initialized to serve as the first Tail

Reply:
The intention is to cross-check that Tail matches the Head. We do have this point of trust and we could also trust both of these headers, but then we introduce an additional assumption that we can prevent.

Requesting the Head is actually optional, as we gonna get the most recent one with gossiping, but it may trigger syncing a bit earlier.

I should rephrase this part as I got the same question in sync.